src/controllers/lift.ts

```diff
@@ -0,0 +1,98 @@
+import express, { Request, Response, NextFunction } from "express";
+import lift, { ILift } from "../models/lift";
+import { EStatus } from "../constants/enums";
+import LiftModel from "../models/lift";
+import mongoose from "mongoose";
+import Joi from "joi";
+
+const LiftValidation = Joi.object({
+  name: Joi.string().required(),
+  elevation_gain: Joi.number().min(1).max(100).required(),
+  status: Joi.string().valid("HOLD", "OPEN", "CLOSED").required(),
+});
+
+export const CreateLift = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const { error, value } = LiftValidation.validate(req.body);
+    if (error) {
+      return res
+        .status(400)
+        .json(` Lift cannot be created due to error : ${error}`);
+    }
+    //console.log('content of value : ', value )
+    const { name, elevation_gain, status } = value;
+    const newLift = {
+      name: name,
+      elevation_gain: elevation_gain,
+      status: status,
+    };
+    const createdLift = await LiftModel.create(newLift);
+    res.status(201).json(`New Lift Created `);
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: "Server error" });
+  }
+};
+export const GetAllLift = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const Lifts = await LiftModel.find();
+    if (Lifts.length === 0 || !Lifts) {
```

Comment:
Hamza we have already discuss this. That there would be no branching in our main code and if we repeat these mistakes them we will dace issue with clients and client always take harsh decisions